core/src/main/java/com/linecorp/armeria/server/file/AbstractHttpFile.java

```diff
@@ -258,31 +256,26 @@
                 // See https://datatracker.ietf.org/doc/html/rfc7232#section-6 for more information
                 // about how conditional requests are handled.
 
-                // Handle 'if-none-match' header.
+                // Handle 'if-none-match', 'if-match' and 'if-modified-since' header.
+                // 'if-match' will probably never be used for GET/HEAD requests
```

Comment:
A partial download can use `if-match` with GET requests, though we haven't implemented it yet. 
https://developer.mozilla.org/en-US/docs/Web/HTTP/Conditional_requests#integrity_of_a_partial_download
How about returning `412 Precondition Failed` for a range request until we support it.